weaviate/collection/data.py

```diff
@@ -270,17 +250,45 @@ def insert(
 
         return self._insert(weaviate_obj)
 
-    def insert_many(self, objects: List[DataObject]) -> List[Union[uuid_package.UUID, Errors]]:
-        weaviate_objs: List[Dict[str, Any]] = [
-            {
-                "class": self.name,
-                "properties": self._parse_properties(obj.data),
-                "id": str(obj.uuid) if obj.uuid is not None else str(uuid_package.uuid4()),
-            }
+    def insert_many(self, objects: List[DataObject]) -> _BatchReturn:
+        weaviate_objs: List[weaviate_pb2.BatchObject] = [
+            weaviate_pb2.BatchObject(
+                class_name=self.name,
+                vector=obj.vector if obj.vector is not None else None,
+                uuid=str(obj.uuid) if obj.uuid is not None else str(uuid_package.uuid4()),
+                properties=self.__parse_properties_grpc(obj.data),
+                tenant=self._tenant,
+            )
             for obj in objects
         ]
 
-        return self._insert_many(weaviate_objs)
+        errors = self._batch.batch(weaviate_objs)
+        if len(errors) == 0:
+            return _BatchReturn(
+                uuids={i: obj.uuid for i, obj in enumerate(weaviate_objs)},
+                errors={},
+                has_errors=False,
+                all_responses=[uuid_package.UUID(obj.uuid) for i, obj in enumerate(weaviate_objs)],
+            )
+
+        return_errors: Dict[int, Error] = {}
+        return_success: Dict[int, uuid_package.UUID] = {}
+        for i, obj in enumerate(weaviate_objs):
+            if i in errors:
+                return_errors[i] = Error(errors[i], original_uuid=objects[i].uuid)
+            else:
+                return_success[i] = uuid_package.UUID(obj.uuid)
+        return _BatchReturn(
+            uuids=return_success,
+            errors=return_errors,
+            has_errors=len(errors) > 0,
+            all_responses=[
+                uuid_package.UUID(obj.uuid)
+                if i not in errors
+                else Error(errors[i], original_uuid=objects[i].uuid)
+                for i, obj in enumerate(weaviate_objs)
+            ],
+        )
```

Comment:
I reckon getting all these `for i, obj in enumerate(weaviate_objs)` into a single loop would be beneficial from a performance PoV. I have something specific in mind so I can push a change here for you to review to speed up the process 😁

Reply:
please push